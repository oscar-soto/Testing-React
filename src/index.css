html, body {
  padding: 70px;
  color: #FFF;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  background-color: #21232A;
}

button {
  padding: 10px 20px;
  border-radius: .25rem;
  border: none;
  font-size: 1.25rem;
  transition: all ease-in-out 0.3s;
  cursor: pointer;
  margin-right: 10px;
}
button:hover {
  opacity: 0.8;
}