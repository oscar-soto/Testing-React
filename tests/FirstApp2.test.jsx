import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Prueba en <FirsApp />', () => {
  const title = 'Assetplam';
  const subTitle = '';

  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje "Assetplam"', () => {
    render(<FirstApp title={title} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Debe de mostrar el titulo en un h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('Debe de mostrar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
