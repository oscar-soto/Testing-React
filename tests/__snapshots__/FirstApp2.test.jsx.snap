// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirsApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy Oscar
     
  </h1>
  <p>
    123
  </p>
  <p>
    123
  </p>
  <p>
    Oscar Espinoza
  </p>
</div>
`;
